core/frontend/src/render/ScreenSpaceEffectBuilder.ts

```diff
@@ -0,0 +1,204 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Rendering
+ */
+
+import { Viewport } from "../Viewport";
+
+/** The underlying data types that can be used for uniform variables in screen-space effect shaders.
+ * @see [[ScreenSpaceEffectBuilder.addUniform]] to define a uniform variable.
+ * @see [[Uniform]] to set the value of a uniform variable.
+ * @beta
+ */
+export enum UniformType {
```

Comment:
probably fine for now, but if we're exposing some GLSL (I assume that's correct for webgl) building API, do these generic pieces need to be in the ScreenSpaceEffectBuilder? Or are we fairly certain that we will not allow other shader building anywhere else that might reuse these enums?

Reply:
You mean, in the same file as ScreenSpaceEffectBuilder? No, they don't have to be in there and they will be moved if we ever use them for other purposes. But I lost the fight for "one export per file" a long time ago.

Reply:
I feel like for such a large file and separate concept two appropriately placed files is better than 1, admittedly 1 export per file is extreme, but either way resolve this conversation if we're deciding to wait until reuse to pull it out.

Reply:
If 204 lines is "large" then I wonder what adjective you'd use for ViewTool.ts' 4477 lines. :D I do take your point but since the documentation of these types heavily references the only thing they're currently used for (ScreenSpaceEffectBuilder) I'll leave them in this file for the time being.

Reply:
Well I meant large contextually... I'm just extra nit-picky on stylistic problems in code I don't have to change myself :P, but like I said I can tell this is better just resolving. I'll resolve it.